File: LP/c/2.c
#include <stdio.h>

int main(){

    int num;
    printf("informe um numero: ");
    scanf("%d", &num);

    for (int i = 2; i <= num; i++){

        if((num % i) == 0 && i < num){
            printf("não é primo");
            break;
        }else{printf("é primo"); break;}
    }
    
    return 0;
}

File: LP/haskell/2.hs
main :: IO ()

primo :: Int -> Bool
primo n
  | n <= 1    = False
  | otherwise = null [x | x <- [2..isqrt n], n `mod` x == 0]    --gera uma lista de 2 até a raiz do numero
                                                                --e ve se o numero é divisivel, se não for retorna true
  where
    isqrt = floor . sqrt . fromIntegral
    
main = return ()